jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -25,9 +23,9 @@ public void onAttach(Context context) {
 
         // Retrieve the main activity.
         try {
-            main_activity = (MainActivity) context;
+            mainActivity = (MainActivity) context;
```

Comment:
A take away here is that a `Fragment` is born with a reference to the owning `Activity`. We will use this when we look at the fragment to activity, or other fragment, callback pattern we discussed earlier.